File: python/api-fetch-prices/src/fetch-prices.py
import requests
import json
import os

API_KEY = os.environ("AV_API_KEY")
SYMBOL = "ICE"

def fetch_prices(api_key, symbol):
  url = f"https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol={SYMBOL}&interval=1min&apikey={API_KEY}"
  response = requests.get(url)
  response.raise_for_status()
  data = json.loads(response.text)
  time_series = data['Time Series (1min)']
  latest_timestamp, latest_data = next(iter(time_series.items()))
  price = latest_data['4. close']
  return price

if __name__ == "__main__":
  symbol = "ICE"

  price = fetch_prices(api_key, symbol)
  print(f"Here is the current price for {symbol}: ${price}")

File: javascript/api-fetch-prices/src/fetch-prices.js
const apiKey = 'TODO-API-KEY';
const symbol = 'ICE';

// change to a const after testing
function fetchStockPrice(apiKey, symbol) {
  const url = `https://www.alphavantage.co/query?function=TIME_SERIES_INTRADAY&symbol=${symbol}&interval=1min&apikey=${apiKey}`;

  fetch(url)
    .then(response => {
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
      return response.json();
    })
    .then(data => {
      const timeSeries = data['Time Series (1min)'];
      const latestTimestamp = Object.keys(timeSeries)[0];
      const latestData = timeSeries[latestTimestamp];
      const price = latestData['4. close'];
      console.log(`The current price of ${symbol} is: $${price}`);
    })
    .catch(error => {
      console.error('Error fetching stock price:', error);
    });
}


fetchStockPrice(apiKey, symbol);
